README.md

```diff
@@ -1,6 +1,23 @@
 # consul-watch-k8s-example
 
-## Using Consul Watch to track config changes in Consul KV store
+## Using Consul Watch to track config changes in Consul KV (Key/Value) store
+
+Tracking KV changes using watches can have two approaches:
+a. Configuring Watches on the Consul Agent that is running as Daemonset.
+b. Configuring Watches on the Consul Agent that is running as a sidecar.
+
+## a. Configuring watches on the Consul Agent running as a Daemonset.
+
+In this scenario, Consul Agent runs as a Daemonset on each node. Consul Agent is configured with watches that tracks KV changes for the Apps/Pods located on that particular node. Generally, watches are configured using CLI command `consul watch` or using json file placed in config directory of Consul Agent. Consider these watches use http handler to trigger notification of KV changes. Each watch will require an IP address of respective App/Pods to notify the KV changes. This scenario works fine until all the Pods are running perfectly. But, as pods are ephemeral, there is a possibility of Pod going down. In such a case, this scenario fails to work as intended.
+
+Suppose node 1 has Pod A for which a watch is configured to track KV-A and it uses http handler pointing to the App/Pod's IP (say 1.1.1.1) to notify KV changes. But if POD A terminates and is scheduled on node 2 with a new IP in such a case, watch configured on node 1 will not be able to trigger or connect to POd A due to changed IP and the fact that watches cannot be configured on the fly as it is configured only through CLI or using config file.
```

Comment:
Should we try to elaborate more on the issue related to on the fly configuration? Like more details about how a newly created Pod won't be able to update the watches easily (not straight forward way to do so)

Reply:
Yes, we should elaborate more on that issue. Will include the related write up.